trinity/config.py

```diff
@@ -468,3 +468,7 @@ def node_class(self) -> Type['Node']:
     def process_id_file(self, process_name: str):  # type: ignore
         with PidFile(process_name, self.data_dir):
             yield
+
+
+class TrinityBeaconConfig(TrinityConfig):
```

Comment:
I'd rather not do derivative/inheritance base configurations.  What do you think about instead having some concept of sub-configs and an API something like:

- `TrinityConfig` -> Base configuration, only handles things that are truly *global*
- mechanism for adding *sub-config* classes which the `TrinityConfig` would be responsible for instantiating as part of the `TrinityConfig.__init__`
- Each sub-config class would have an API for initializing itself from the cli args: `BaseSubConfig.from_cli_args(cls, trinity_config, cli_args)`

Reply:
Yep, sounds good. Will give it a shot. Maybe I move Beacon node support into a follow up PR because this is mostly finished apart from that.

Reply:
I think your proposal has some nice properties that I just became aware of so I'm posting them here.

So, initially it may seem like having a `BaseSubConfig` and then derived classes from there doesn't buy us much from subclassing a `BaseTrinityConfig` directly. However the benefits become more obvious when we think about the possible use cases.

E.g Let's say we would have a `BaseTrinityConfig` and then another `Eth1TrinityConfig` for the core Ethereum 1 configs. Now lets say we want to add configs specific to light mode and lets assume they would be reusable across Ethereum 1 and 2. In this model, we would be forced to derive both `Eth1TrinityLightConfig` and `Eth2TrinityLightConfig` and duplicate the light node settings.

In constrast, with the proposed model, it works more like a mixin where we can just reuse the `LightSubConfig` to a compositon of `TrinityConfig`+ `Eth1SubConfig` + `LightSubConfig` as well as `TrinityConfig` + `Eth2SubConfig` + `LightSubConfig`.

So, when I first thought about this, I thought that straight top level inheritance would be just fine here as we are mostly talking about data with barely any behavior. But turns out, this has some nice extra properties (it also has some other nice properties) that just became obvious to me now.

So, thanks for pointing it out @pipermerriam